plutus-core/testlib/PlutusCore/Generators/QuickCheck/Builtin.hs

```diff
@@ -125,6 +136,63 @@ instance ArbitraryBuiltin ByteString where
     arbitraryBuiltin = Text.encodeUtf8 <$> arbitraryBuiltin
     shrinkBuiltin = map Text.encodeUtf8 . shrinkBuiltin . Text.decodeUtf8
 
+-- | Generate a tag for the 'Constr' constructor.
+genConstrTag :: Gen Integer
+genConstrTag = frequency
+    [ -- We want to generate most plausible constructor IDs most often.
+      (6, chooseInteger (0, 2))
+    , -- Less plausible -- less often.
+      (3, chooseInteger (3, 5))
+    , -- And some meaningless garbage occasionally just to have good coverage.
+      (1, abs <$> arbitraryBuiltin)
```

Comment:
Because otherwise even serialization-deserialization roundtrip fails (because we can serialize a negative `Constr` tag, but can't deserialize it) and we want to keep that test.